src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -142,24 +200,22 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
     void ActionMap::_PopulateAvailableActionsWithStandardCommands(std::unordered_map<hstring, Model::ActionAndArgs>& availableActions, std::unordered_set<InternalActionID>& visitedActionIDs) const
     {
         // Update AvailableActions and visitedActionIDs with our current layer
-        for (const auto& [actionID, cmd] : _ActionMap)
+        for (const auto& [_, cmd] : _ActionMap)
         {
-            if (cmd.ActionAndArgs().Action() != ShortcutAction::Invalid)
+            // Only populate AvailableActions with actions that haven't been visited already.
+            const auto actionID = Hash(cmd.ActionAndArgs());
+            if (visitedActionIDs.find(actionID) == visitedActionIDs.end())
             {
-                // Only populate AvailableActions with actions that haven't been visited already.
-                if (visitedActionIDs.find(actionID) == visitedActionIDs.end())
+                const auto& name{ cmd.Name() };

```

Comment:
I still don't think we should use `const auto&` locals. It's hard to read because capturing a temporary/return value with `&` _usually_ leads to a crash.

e.g. this will cause a crash at runtime:

```c++
const hstring& name{ cmd.Name() };
```

and this will not:

```c++
const auto& name{ cmd.Name() };
```

subtle, eh? that's one of the weird things about `auto` - it just ignores your `&` in this case.